src/enums.h

```diff
@@ -314,9 +314,37 @@ inline bool operator<( const tripoint &a, const tripoint &b )
 
 static constexpr tripoint tripoint_min { INT_MIN, INT_MIN, INT_MIN };
 static constexpr tripoint tripoint_zero { 0, 0, 0 };
+static constexpr tripoint tripoint_max{ INT_MAX, INT_MAX, INT_MAX };
 
 static constexpr point point_min{ tripoint_min.x, tripoint_min.y };
 static constexpr point point_zero{ tripoint_zero.x, tripoint_zero.y };
+static constexpr point point_max{ tripoint_max.x, tripoint_max.y };
+
+/** Checks if given tripoint is inbounds of given min and max tripoints using given clearence **/
+inline bool generic_inbounds( const tripoint &p,
+                              const tripoint &p_min = tripoint_min,
+                              const tripoint &p_max = tripoint_max,
+                              const tripoint &clearence_min = tripoint_zero,
+                              const tripoint &clearence_max = tripoint_zero )
```

Comment:
To reduce the number of arguments we could have a type for a box (rectangle) which is stored as two tripoints (points) for the min and max corners of the box.

Not sure whether this single use warrants that, but wanted to raise the possibility.

Reply:
Good point.